# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

## My Lane: Lane 1 - Ranking Signal Analysis

For this assignment, I have provisionally selected **Lane 1 - Ranking Signal Analysis**.

I chose this lane because I want to understand how observable content and search performance signals relate to content visibility and engagement.

Before building predictive machine learning models, it is important to first identify which measurable factors appear to have meaningful relationships with search performance.

This lane allows me to explore the dataset, study important search signals, and perform evidence-based analysis instead of making assumptions.

The insights gained from this analysis are expected to provide a strong foundation for future machine learning work. Since the lane can be changed until Week 4, I will continue evaluating whether it remains the best fit as I learn more about the data.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

## Research Question

### Search Question
Which observable content and search performance signals are most strongly associated with higher content visibility and engagement?

### Unit of Analysis
The unit of analysis is a single content page (one row in the starter dataset). Each page contains measurable attributes such as impressions, clicks, CTR, content age, engagement, and other observable search-related metrics.

### Decision
The goal is to help SEO analysts and content teams understand which observable signals should receive more attention when evaluating content performance.

### Output
The expected output is an evidence-based summary that identifies the observable signals most closely associated with stronger or weaker search performance.

### Action
The findings can help analysts prioritize further investigation, identify content that may require optimization, and support future data-driven content improvement decisions.

### Cost of a Wrong Recommendation
If the analysis incorrectly identifies important signals, teams may spend time optimizing content based on weak or misleading evidence. On the other hand, overlooking genuinely important signals could result in missed opportunities to improve content performance and search visibility.

### Why Data Analysis Helps
Large websites often contain thousands of content pages and many performance metrics. Examining these pages manually is difficult and time-consuming. Data analysis helps identify meaningful patterns within the available data and supports more informed decision-making before developing predictive machine learning models.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [18]:
import pandas as pd

# Load the starter dataset
df = pd.read_csv("../../data/raw/content_refresh_anonymized.csv")

# Display basic information
print("Dataset Shape:", df.shape)

print("\nColumn Names:")
print(df.columns.tolist())

Dataset Shape: (30000, 44)

Column Names:
['content_id', 'client_id', 'search_volume', 'competition', 'competition_level', 'cpc', 'content_type', 'main_intent', 'word_count', 'char_count', 'provider_used', 'model_used', 'impressions_90d', 'clicks_90d', 'pageviews_90d', 'sessions_90d', 'users_90d', 'engaged_sessions_90d', 'ai_sessions_90d', 'scroll_events_90d', 'days_with_impressions', 'days_with_sessions', 'impressions_last_30d', 'clicks_last_30d', 'sessions_last_30d', 'impressions_prev_30d', 'clicks_prev_30d', 'sessions_prev_30d', 'content_age_days', 'age_tier', 'age_tier_order', 'days_since_last_update', 'freshness_tier', 'word_count_tier', 'char_count_tier', 'ctr', 'avg_position', 'engagement_rate', 'scroll_rate', 'ai_traffic_pct', 'impression_tier', 'position_tier', 'trend_direction', 'trend_pct']


In [19]:
# Basic dataset information
print(f"Number of rows: {df.shape[0]}")
print(f"Number of columns: {df.shape[1]}")

Number of rows: 30000
Number of columns: 44


In [20]:
# Three useful statistics
print(f"Average 90-day impressions: {df['impressions_90d'].mean():.2f}")
print(f"Average CTR: {df['ctr'].mean():.2f}%")
print(f"Average content age (days): {df['content_age_days'].mean():.2f}")

Average 90-day impressions: 5200.37
Average CTR: 0.51%
Average content age (days): 256.17


In [21]:
# Distribution of trend direction
print(df["trend_direction"].value_counts())

trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64


## Quick Look at the Data

The starter dataset contains **30,000 content pages** with **44 measurable features**, providing a rich set of content and search performance signals for analysis.

Some key observations from the starter dataset are:

- The average **90-day impressions** are **5,200.37**.
- The average **CTR** is **0.51%**.
- The average **content age** is **256.17 days**.

The dataset also contains metrics such as clicks, sessions, engagement rate, average position, content freshness, and trend direction. These variables are directly relevant to understanding relationships between observable signals and search performance.

These statistics indicate that the dataset contains sufficient variation in visibility, engagement, and content freshness. This supports my choice of **Lane 1 – Ranking Signal Analysis**, as the available data provides a strong foundation for investigating which observable signals are associated with better search performance.

Additionally, The trend distribution also shows pages in multiple categories (up, down, stable, new, and flat), indicating meaningful variation in content performance that can be explored further.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

## Careful Words

This project is an exploratory analysis of the available starter dataset. The goal is to identify observable relationships between content characteristics and search performance metrics.

The analysis may identify patterns and associations that can support decision-making, but it cannot prove that one factor directly causes another. Any observed relationship should be interpreted as evidence for further investigation rather than proof of causation.

The findings will be limited to the available dataset and its measured variables. They should not be interpreted as predictions of Google's ranking algorithm or as universally applicable SEO rules.

The purpose of this work is to provide data-informed insights that can guide future analysis and machine learning, rather than making definitive claims about search rankings.

## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.